resources/assets/coffee/_classes/store-supporter-tag.coffee

```diff
@@ -0,0 +1,189 @@
+      Math.max(0, Math.round(raw, 0))
+  }
+
+  constructor: ->
+    $(document).on 'turbolinks:load', =>
+      @initialize(document.getElementById('js-store-supporter-tag'))
+
+  initialize: (rootElement) =>
+    return unless rootElement
+    @el = rootElement
+    @searching = false
+    @searchData = null
+    @currentUser = {
+      username: @el.dataset.username
+      avatar_url: @el.dataset.avatarUrl
+    }
+    # Everything should be scoped under the root @el
+    @priceElement = @el.querySelector('.js-price')
+    @durationElement = @el.querySelector('.js-duration')
+    @pricePerMonthElement = @el.querySelector('.js-price-per-month')
+    @discountElement = @el.querySelector('.js-discount')
+    @slider = @el.querySelector('.js-slider')
+    @sliderPresets = @el.querySelectorAll('.js-slider-preset')
+    @usernameInput = @el.querySelector('#username.form-control')
+
+    @debouncedGetUser = _.debounce @getUser, 300
+    @initializeSlider()
+    @initializeSliderPresets()
+    @initializeUsernameInput()
+    @
+
+  initializeSlider: =>
+    slider = $(@slider).slider {
+      range: 'min',
+      value: @sliderValue(@MIN_VALUE),
+      min: @sliderValue(@MIN_VALUE),
+      max: @sliderValue(@MAX_VALUE),
+      step: 1,
+      animate: true,
+      slide: (event, ui) =>
+        @onSliderValueChanged event, ui
+      change: (event, ui) =>
+        @onSliderValueChanged event, ui
+    }
+    @updatePrice(@calculate(@sliderValue(@MIN_VALUE)))
+    slider
+
+  initializeSliderPresets: =>
+    $(@sliderPresets).on 'click', (event) =>
+      target = event.currentTarget
+      price = @CUTOFFS[target.dataset.months]
+      $(@slider).slider('value', @sliderValue(price)) if price
+
+  initializeUsernameInput: =>
+    $(@usernameInput).on 'input', (event) =>
+      @onInput(event)
+
+  getUser: (username) =>
+    $.post '/users/check-username-exists', username: username
+    .done (data) =>
+      @setUserInteraction(data?)
+      @updateUserDisplay(data)
+      @updateCart(data)
+    .fail (xhr) =>
+      @setUserInteraction(false)
+      @updateUserDisplay(null)
+      @updateCart(null)
+      if xhr.status == 401
+        osu.popup osu.trans('errors.logged_out'), 'danger'
+    .always =>
+      @searching = false
+
+  calculate: (position) =>
+    price = Math.floor(position / @RESOLUTION)
+    duration = switch
+      when price < 8 then 1
+      when price < 12 then 2
+      when price < 16 then 4
+      when price < 20 then 6
+      when price < 22 then 8
+      when price < 24 then 9
+      when price < 25 then 10
+      when price < 28 then 10
+      else
+        Math.floor(price / 26.0 * 12)
+
+    Object.assign(Object.create(StoreSupporterTag.Price), { price: price, duration: duration })
+
+  onSliderValueChanged: (event, ui) =>
+    values = @calculate(ui.value)
+    @updatePrice(values)
+    @updateSliderPresets(values)
+
+  onInput: (event) =>
+    if !@searching
```

Comment:
it's not supposed to `return if @searching`; it's a stupid hack to force a render because it might be delayed until after the debounce triggers.

which is a mystery